latch_cli/docker_utils/__init__.py

```diff
@@ -0,0 +1,212 @@
+                            apt-get install -y software-properties-common && \\
+                            add-apt-repository "deb http://cloud.r-project.org/bin/linux/debian buster-cran40/" && \\
+                            apt-get install -y r-base r-base-dev libxml2-dev libcurl4-openssl-dev libssl-dev wget \
+                        """
+                    ).strip(),
+                    "copy environment.R /root/environment.R",
+                    "run Rscript /root/environment.R",
+                ],
+                order=DockerCmdBlockOrder.PRECOPY,
+            )
+        )
+
+    # conda environment in workflow directory
+    conda_env_file_name = None
+    if (pkg_root / "environment.yml").exists():
+        conda_env_file_name = "environment.yml"
+    elif (pkg_root / "environment.yaml").exists():
+        conda_env_file_name = "environment.yaml"
+
+    if conda_env_file_name is not None:
+        print("Create conda environment from `environment.yml`")
+        with open(pkg_root / conda_env_file_name) as f:
+            conda_env = yaml.safe_load(f.read())
+            if "name" in conda_env:
+                env_name = conda_env["name"]
+            else:
+                env_name = "unnamed"
+
+        commands += [
+            DockerCmdBlock(
+                comment="Set conda environment variables",
+                commands=[
+                    "env CONDA_DIR /opt/conda",
+                    "env PATH=$CONDA_DIR/bin:$PATH",
+                ],
+                order=DockerCmdBlockOrder.PRECOPY,
+            ),
+            DockerCmdBlock(
+                comment="install miniconda",
+                commands=[
+                    dedent(
+                        """\
+                        run apt-get update -y && \\
+                            apt-get install -y curl && \\
+                            curl -O https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh && \\
+                            mkdir /root/.conda && \\
+                            bash Miniconda3-latest-Linux-x86_64.sh -b -p /opt/conda && \\
+                            rm -f Miniconda3-latest-Linux-x86_64.sh && \\
+                            conda init bash \
+                        """
+                    ).strip(),
+                ],
+                order=DockerCmdBlockOrder.PRECOPY,
+            ),
+            DockerCmdBlock(
+                comment="build and configure conda environment",
+                commands=[
+                    f"copy {conda_env_file} /root/{conda_env_file}",
+                    f"run conda env create -f {conda_env_file} --name {env_name}",
+                    f"""shell ["conda", "run", "-n", "{env_name}", "/bin/bash", "-c"]""",
+                    "run /opt/conda/bin/pip install --upgrade latch",
+                ],
+                order=DockerCmdBlockOrder.PRECOPY,
+            ),
+        ]
+
+    # python package in workflow directory (poetry, distutils, etc.)
```

Comment:
```suggestion
    # poetry, distutils, etc.
```
this isn't actually just poetry (and poetry uses a different non-standard approach) - there is a PEP for this somewhere that you probably should mention instead